cmd/docker/builder.go

```diff
@@ -73,6 +73,12 @@ func processBuilder(dockerCli command.Cli, cmd *cobra.Command, args, osargs []st
 		return args, osargs, nil
 	}
 
+	// wcow build command must use the legacy builder
+	// if not opt-in through a builder component
+	if !useBuilder && dockerCli.ServerInfo().OSType == "windows" {
+		return args, osargs, nil
+	}
```

Comment:
Repro

```console
# on WSL2
docker buildx bake binary --set *.platform=windows/amd64
```

_Switch to Windows containers_ on Docker Desktop

```powershell
# on powerhell or cmd
> .\build\docker-windows-amd64.exe info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.7.1)
  compose: Docker Compose (Docker Inc., v2.2.3)
  extension: Manages Docker extensions (Docker Inc., v0.0.1-alpha.4)
  hub: Docker Hub (Docker Inc., 346ece73cb-dirty)
  scan: Docker Scan (Docker Inc., v0.17.0)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 4
 Server Version: 20.10.12
 Storage Driver: windowsfilter
  Windows:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local logentries splunk syslog
 Swarm: inactive
 Default Isolation: hyperv
 Kernel Version: 10.0 22000 (22000.1.amd64fre.co_release.210604-1628)
 Operating System: Windows 10 Pro Version 2009 (OS Build 22000.527)
 OSType: windows
 Architecture: x86_64
 CPUs: 32
 Total Memory: 63.93GiB
 Name: pc-sff
 ID: BJ23:IQXR:IXL3:FAEE:TGKX:5XWD:7ZKB:6GXF:P5BL:LU3E:T45O:MULU
 Docker Root Dir: C:\ProgramData\Docker
 Debug Mode: false
 Username: ...
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

```dockerfile
FROM mcr.microsoft.com/windows/nanoserver:ltsc2022
WORKDIR /app
RUN ver
```

```powershell
> docker build .
Sending build context to Docker daemon  2.048kB
Step 1/3 : FROM mcr.microsoft.com/windows/nanoserver:ltsc2022
ltsc2022: Pulling from windows/nanoserver
3ab33c1d9cc1: Pull complete
Digest: sha256:8aa115e244add588e9a71a38d302e8fc801f2e986be83a58d5e015e4cba8f613
Status: Downloaded newer image for mcr.microsoft.com/windows/nanoserver:ltsc2022
 ---> 6dde50c651d0
Step 2/3 : WORKDIR /app
 ---> Running in ed30b4ce8844
Removing intermediate container ed30b4ce8844
 ---> b261db46d93b
Step 3/3 : RUN ver
 ---> Running in ff8390d7dd39

Microsoft Windows [Version 10.0.20348.524]
Removing intermediate container ff8390d7dd39
 ---> f3e0e001378a
Successfully built f3e0e001378a
```

@thaJeztah Side note about what is reported by `docker info` in _Server_ with WCOW:

```
 Operating System: Windows 10 Pro Version 2009 (OS Build 22000.527)
```

This one is wrong. Should be smth like:

```
 Operating System: Windows 11 Pro Version 21H2 (OS Build 22000.527)
```

Reply:
@crazy-max could you open a ticket for that in https://github.com/moby/moby ?

Reply:
yes sure